core/src/Streamly/Internal/Data/Ring.hs

```diff
@@ -137,7 +147,7 @@ advance rb ringHead =
 -- array.
 {-# INLINE moveBy #-}
 moveBy :: Int -> Ring a -> Int -> Int
-moveBy by rb ringHead = (ringHead + by) `mod` ringCapacity rb
+moveBy by rb ringHead = (ringHead + by) `mod1` ringCapacity rb
```

Comment:
usually we should not be moving it by more than the ring size. But we have to handle the case if someone does that. Anyway, it seems this function is not used anywhere.